.circleci/config.yml

```diff
@@ -0,0 +1,126 @@
+            - stack-root-{{ arch }}-ghc8.6.3-
+            - "stack-root-v2"
+
+      - restore_cache:
+          keys:
+            - toolchain-{{ arch }}-{{ epoch }}
+            - toolchain-{{ arch }}-
+            - "toolchain-v2"
+
+      - restore_cache:
+          keys:
+            - homebrew-{{ epoch }}
+            - homebrew-
+            - homebrew
+
+      - run:
+          name: Setup Toolchain
+          command: |
+            if ! [ -x $HOME/.ghcup/bin/ghcup ]; then
+              mkdir -p $HOME/.ghcup/bin
+              cd $HOME/.ghcup/bin
+              curl -LO https://github.com/haskell/ghcup/releases/download/0.0.7/ghcup
+              chmod +x $HOME/.ghcup/bin/ghcup
+            fi
+
+            export PATH="$HOME/.cabal/bin:$HOME/.ghcup/bin:$PATH"
+
+            ghcup set "8.6.3" || ghcup install "8.6.3" && ghcup set "8.6.3"
+            command -v cabal || ghcup install-cabal "2.4.1.0"
+
+            if ! [ -x $HOME/.local/bin/stack ]; then
+              curl -sSL https://get.haskellstack.org/ | sh
+            fi
+
+            stack config set system-ghc --global true
+            stack config set install-ghc --global false
+
+            echo 'export PATH="$HOME/.cabal/bin:$HOME/.ghcup/bin:$PATH"' >> $BASH_ENV
+
+      - save_cache:
+          key: toolchain-{{ arch }}-{{ epoch }}
+          paths:
+            - "~/.ghcup"
+
+      - run:
+          name: Homebrew Dependencies
+          command: |
+            brew install coreutils
+            if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+              brew tap caskroom/cask
+              brew cask install google-cloud-sdk
+            fi
+
+      - save_cache:
+          key: homebrew-{{ epoch }}
+          paths:
+            - /usr/local/Homebrew
+
+      - run:
+          name: Build Dependencies
+          command: stack build --no-terminal --dependencies-only
+
+      - save_cache:
+          key: stack-root-{{ arch }}-ghc8.6.3-{{ checksum "snapshot.yaml" }}-{{ checksum "package.yaml" }}
+          paths:
+            - "~/.stack"
+
+      - run:
+          name: Build Executables
+          command: |
+            cabal v1-update
+            cabal v1-configure \
+              --prefix=/usr/local \
+              --docdir=/usr/local/share/doc/radicle \
+              --libsubdir=radicle \
+              --datasubdir=radicle \
+              --libexecsubdir=radicle \
+              --package-db="$(stack path --snapshot-pkg-db)" \
+              --package-db="$(stack path --local-pkg-db)" \
+              --package-db="$(stack path --global-pkg-db)"
+            cabal v1-build
+
+      - run:
+          name: Package
+          command: |
+            if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+              cabal v1-copy \
+                exe:rad \
+                exe:radicle \
+                exe:radicle-daemon \
+                exe:rad-machines \
+                --destdir=dist
```

Comment:
How about using `dist/package-root`? I am a bit confused by putting the tarball in the package root next to `usr`.

Reply:
Do you mean `package-root` instead of `pkg`? The tarball is in `dist/pkg`

Reply:
I mean using `--destdir=dist/package-root`, then `tar -C dist/package-root -cvzf dist/pkg/$tarball .`. Otherwise we end up with `dist/usr` and `dist/pkg` next to each other which I find confusing.